src/Lesson5/ShopZD2.java

```diff
@@ -0,0 +1,72 @@
+package Lesson5;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+
+public class ShopZD2 implements Shop {
+
+    private Map<Product, Integer> mapProduct = new HashMap<>();
+    private Check check = new Check();
+    private Map<Integer,Product> productSearch=new HashMap<>();
+
+    public ShopZD2() {
+        super();
+    }
+
+    public Check generateCheckProduct(List<Integer> requiredProductIds) {
+        for (int i = 0; i < requiredProductIds.size(); i++) {
+         copyingListToCheck(requiredProductIds.get(i));
+        }
+        return check;
+    }
+
+
+    private void copyingListToCheck(int requiredProductIds){
```

Comment:
![image](https://user-images.githubusercontent.com/13934697/68968882-a7643b80-07f4-11ea-8d78-a8dc336b343e.png)

Reply:
you are calling one method 6 times!!!!!!!!!!!!!!